[WIP] Add support for parsing word fractions / compound amounts

karlentwistle opened the pull request:
global-web was incorrectly assigning 'half' as a unit of measurement. A unit of measurement is supposed to be 'grams', or 'ounces' or 'stone' etc etc. Half is an amount that's been represented in words. This PR adds support for expressing fractions as words - 'wordy fractions'.

- [x] Make this an optional feature (just don't include either of the translations)
- [ ] Refactor specification 
- [ ] Fix container size business logic incorrectly mutating 'half a tsp salt' to become '1/2 (1tsp) salt'
- [ ] Test how do those improvements do in other languages
- [ ] Add nice screen capture to the PR description with some examples

JuanitoFatas commented:
Q: Should this be a optional feature? So that if people in region prefer words won't be converted to fractional.

balvig commented:
@karlentwistle thanks for this! 

One thing I'm wondering about: 

Do we have a reason to treat this separately from `word_digits`? Ie if for example I went and added

```yml
:numbers:
   quarter: 1/4
```

...to the existing implementation, would I get a similar result? (I'm guessing "no", but it's a little tricky to detect from the current spec 🙇  Perhaps it would help with a usage spec like [this one]( https://github.com/cookpad/ingreedy/blob/master/spec/ingreedy_spec.rb#L20-L25)?)

balvig commented:
(incidentally, just noticed this: https://github.com/cookpad/global-web/blob/1bf57b39686f4a15e4d8f424e2f61c6a91e2e052/config/ingreedy/id.yml#L531-L538 🤔 )

davidstosik commented:
Maybe worth pasting my Slack comments here:

Looks pretty good! It's nice to have more options to input quantities, for a seamless experience! :thumbsup:
I have two questions:
- is `1/2 (1 cup)` a standard way to display half a cup? Is there a chance users may misunderstand that syntax? (I would think it'd be easier to understand with `1/2 cup` or `0.5 cup`)
- how do those improvements do in other languages?

karlentwistle commented:
@balvig it does have the same affect as adding to numbers - the only reason I hesitated to do this was because right now all numbers (apart from in Indonesia) conform to http://ruby-doc.org/core-2.1.0/Numeric.html but the word fractions are definitely http://ruby-doc.org/core-2.1.0/String.html - it feels like this could be annoying in the future if we want to do any Math based operations on the amounts.

Also a next logical step for ingreedy is probably to have 'compound wordy fraction' - so that `one and a half tsp salt` parses correctly like `1 1/2 tsp salt` already does. I think this might be easier to achieve having `numbers` and `word_fractions` discreetly separated.

@davidstosik I've changed this PR to [WIP] since I just wanted to gather some early feedback - I will add a nice screen capture to the PR description soon! 😄 

> is 1/2 (1 cup) a standard way to display half a cup? Is there a chance users may misunderstand that syntax? I would think it'd be easier to understand with 1/2 cup or 0.5 cup)

Yes I think it needs to be displayed as 1/2 cup or 0.5 cup unfortunately the container size business logic is coming into affect and causing the unwanted mutation - I've added that as a todo

> how do those improvements do in other languages

I need to do some more research into this (unfortunately English is my native / single language) so might need some help 🤔
